src/functional/ffront/past_to_itir.py

```diff
@@ -178,37 +180,38 @@ def _construct_itir_out_arg(
                 args=[self._construct_itir_out_arg(el) for el in node.elts],
             )
         else:
-            raise AssertionError(
+            raise GTTypeError(
                 "Unexpected `out` argument. Must be a `past.Name`, `past.Subscript`"
-                " or a `past.TupleExpr` of  `past.Name`, `past.Subscript` or`past.TupleExpr` (recursively)."
+                " or a `past.TupleExpr` thereof."
             )
 
     def _construct_itir_domain_arg(
         self,
         out_field: past.Name,
-        node_domain: Optional[past.Dict],
+        node_domain: Optional[past.Expr],
         slices: Optional[list[past.Slice]] = None,
-    ):
+    ) -> itir.FunCall:
         domain_args = []
 
         out_field_types = type_info.primitive_constituents(out_field.type).to_list()
+        out_field_types0_dims = cast(ts.FieldType, out_field_types[0]).dims
```

Comment:
Could this be named more clearly? Maybe something like `first_out_field_type_dims`  ?

Reply:
that is an even longer name...